services/ml/compilation_delegate_dnnl.h

```diff
@@ -0,0 +1,104 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SERVICES_ML_COMPILATION_DELEGATE_DNNL_H_
+#define SERVICES_ML_COMPILATION_DELEGATE_DNNL_H_
+
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "base/macros.h"
+#include "mojo/public/cpp/bindings/strong_binding.h"
+#include "services/ml/common.h"
+#include "services/ml/compilation_impl.h"
+#include "services/ml/public/mojom/model.mojom.h"
+#include "third_party/dnnl/include/dnnl.h"
+
+namespace ml {
+typedef struct {
+  int nargs;
+  dnnl_exec_arg_t* args;
+} args_t;
+
+struct OperationDnnl {
+  explicit OperationDnnl();
+  explicit OperationDnnl(dnnl_primitive_t);
+  explicit OperationDnnl(const mojom::OperationPtr&);
+  ~OperationDnnl();
+  OperationDnnl(const OperationDnnl&);
+
+  dnnl_primitive_t primitive;
+
+  // For custom kernels
+  int32_t type;
+  std::vector<std::string> inputs;
+  std::vector<std::string> outputs;
+};
+
+struct CompiledModelDnnl {
+  CompiledModelDnnl();
+  ~CompiledModelDnnl();
+
+  std::map<std::string, dnnl_memory_t> memories;
+  std::vector<OperationDnnl> operations;
+  std::vector<args_t> operations_args;
```

Comment:
How about moving `std::vector<dnnl_exec_arg_t> args;` to `OperationDnnl`

Reply:
Yes, it's much more convenient.